src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -21,180 +21,180 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
         public TaskCenterSolutionAnalysisProgressReporter(
             SVsTaskStatusCenterService taskStatusCenterService,
             IDiagnosticService diagnosticService,
             VisualStudioWorkspace workspace)
         {
-            _lastTimeReported = DateTimeOffset.UtcNow;
-            _resettableDelay = null;
-
-            ResetProgressStatus();
-
             _taskCenterService = (IVsTaskStatusCenterService)taskStatusCenterService;
-
             _options = new TaskHandlerOptions()
             {
                 Title = ServicesVSResources.Running_low_priority_background_processes,
                 ActionsAfterCompletion = CompletionActions.None
             };
 
-            var crawlerService = workspace.Services.GetService<ISolutionCrawlerService>();
+            var crawlerService = workspace.Services.GetRequiredService<ISolutionCrawlerService>();
             var reporter = crawlerService.GetProgressReporter(workspace);
 
-            StartedOrStopped(reporter.InProgress);
+            if (reporter.InProgress)
+            {
+                Started();
+            }
+            else
+            {
+                Stopped();
+            }
 
-            // no event unsubscription since it will remain alive until VS shutdown
             reporter.ProgressChanged += OnSolutionCrawlerProgressChanged;
         }
 
-        private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progressData)
+        /// <summary>
+        /// Retrieve and throttle solution crawler events to be sent to the progress reporter UI.
+        /// 
+        /// there is no concurrent call to this method since ISolutionCrawlerProgressReporter will serialize all
+        /// events to preserve event ordering
+        /// </summary>
+        /// <param name="progressData"></param>
+        public void OnSolutionCrawlerProgressChanged(object sender, ProgressData progressData)
+        {
+            lock (_lock)
+            {
+                _lastProgressData = progressData;
+
+                // The task is running which will update the progress.
+                if (_intervalTask != null)
+                {
+                    return;
+                }
+
+                // Report progress immediately to ensure we update the UI on the first event.
+                ReportProgress();
+
+                // Kick off task to update the UI after a delay to pick up any new events.
+                _intervalTask = Task.CompletedTask.ContinueWithAfterDelay(() =>
+                {
+                    ReportProgress();
+                }, CancellationToken.None, s_minimumInterval, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+            }
+        }
+
+        private void ReportProgress()
+        {
+            ProgressData data;
+            lock (_lock)
+            {
+                data = _lastProgressData;
+                _intervalTask = null;
+            }
+
+            UpdateUI(data);
```

Comment:
this is still techincally unsafe/racey.  i.e. ReportProgress may be called concurrently on two different threadpool threads.  you either need to ensure that they happen in the right order.  i can discuss ways to do that with you.